### PR TITLE
🌐 Add Chinese translation for `docs/zh/docs/async.md`

### PR DESCRIPTION
**Pull request by jaystone776**
as title

### CODE REVIEW

**Review by zy7y — APPROVED** [review]

**Review by zy7y — APPROVED** [review]

File: docs/zh/docs/async.md [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+
+在 Python 以前的版本中，只能使用线程或 <a href="https://www.gevent.org/" class="external-link" target="_blank">Gevent</a>。但这种代码非常复杂难懂，也不方便调试与理解。
+
+在 NodeJS / 浏览器 JavaScript 以前的版本中要使用“回调”，这种方式直通“<a href="http://callbackhell.com/" class="external-link" target="_blank">回调地狱</a>”。
+
+## 协程
+
+**协程**只是个非常花哨的术语，指的是由 `async def` 函数返回的对象。Python 把它识别为可以在某些点启动或终止的函数，但它还可以在内部暂停 ⏸，只要在它的内部包含 `await`。
+
+但是使用 `async` 与 `await` 的异步编码的这种功能尝尝被统称为“携程”。它与 Go 的核心功能 “Goroutines” 相对应。
```
Comment (mkdir700) [inline docs/zh/docs/async.md:350]: “携程” -> “协程”

```suggestion
但是使用 `async` 与 `await` 的异步编码的这种功能尝尝被统称为“协程”。它与 Go 的核心功能 “Goroutines” 相对应。
```

File: docs/zh/docs/async.md [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+    
+    这些是非常细的技术细节，介绍的是 **FastAPI** 底层运作机制。
+    
+    如果你了解足够的技术知识（协程、线程、阻塞等），并对 FastAPI 如何处理 `async def` 和普通的 `def` 感兴趣，请继续。
+
+### 路径操作函数
+
+使用普通 `def` 替代 `async def` 声明*路径操作函数*时，要在等待的外部线程池中运行，不能直接调用（因为它会阻塞服务器）。
+
+如果你之前使用的异步框架不以上述方式运行，或者你习惯了定义琐碎的仅供计算的纯 `def` *路径操作函数*只为了提升些许性能（约 100 纳秒），请注意，在 **FastAPI** 中的效果正相反。在这些情况下，除非*路径操作函数*使用执行阻塞 <abbr title="输入/输出：磁盘读取与写入，网络通信d。">I/O</abbr> 的代码，最好使用 `async def`。
```
Comment (mkdir700) [inline docs/zh/docs/async.md:376]:
```suggestion
如果你之前使用的异步框架不以上述方式运行，或者你习惯了定义琐碎的仅供计算的普通`def` *路径操作函数*而只为了提升些许性能（约 100 纳秒），请注意，在 **FastAPI** 中的效果正相反。在这些情况下，最好使用 `aysnc def`，除非*路径操作函数*使用阻塞式 <abbr title="输入/输出：磁盘读取与写入，网络通信">I/O</abbr> 的代码。
```

**Review by mkdir700 — CHANGES_REQUESTED** [review]

**Comment by tiangolo** [comment]
📝 Docs preview for commit 99fac02a2d2c9d87a2ad45d6711330b62888f01a at: https://649a2380c3bf901b5aa9e53c--fastapi.netlify.app

**Comment by tiangolo** [comment]
Thank you! It seems this one was handled in another PR, so I'll pass on this one, but feel free to add new PRs updating the current version if you think that's needed!

Thanks! ☕ 🍰

**Pull request closed without merging**